### ISSUE TITLE
Starfield: Creation Kit (2722710)

### ISSUE THREAD

**Issue by Blayer98** [ValveSoftware/Proton#7884]
## System Information
- GPU: AMD AMD Custom GPU 0405 (vangogh, LLVM 15.0.7, DRM 3.54, 6.1.52-valve16-1-neptune-61)
- Video driver version: 4.6 (Compatibility Profile) Mesa 23.1.3 (git-87ebaf765d)
- Kernel Version: 6.1.52-valve16-1-neptune-61
- Link to full system information report: [Gist System Information](https://gist.github.com/Blayer98/f127c9fa4daabec09b34e97a1040042e)
- Proton version: Proton Experimental

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

Proton Log file: 
[steam-2722710.log](https://github.com/user-attachments/files/16169195/steam-2722710.log)

## Symptoms

While the application launches fine (sometimes it takes a couple seconds to load and stays on black windows) a popup says that the graphics drivers are 314 days old upon launch but this can be dismissed. Attempting to save an esp fie (plugin) causes the application to crash and a CreationKit_07-10-21-30.dmp file is created in the same folder as both Starfield and CreationKit's exe files. (the date & time will be the date & time of the system you’re using)

The issue of plugins not being saved is not specific to Steam Deck, according to posts on the Steam Community Forums for Starfield’s Creation Kit, it is happening on Linux generally as well, although I don’t know if this is an issue specific to Proton or not.

## Reproduction

Simply launch the Creation Kit through Steam. If on Steam Deck like me, you should also get the same graphics driver popup. Once the app has properly loaded, simply attempt to save a new plugin. The application should crash.

**Comment by kisak-valve** [comment]
Hello @Blayer98, `wine: Call from 00006FFFFFFA2DC8 to unimplemented function KERNEL32.dll.CopyFile2, aborting` looks like the line of interest from your Proton log.

**Comment by Blayer98** [comment]
Ok, that’s very weird, does Wine not have KERNEL32 fully implemented? If so then I’m not sure if any potential fixes for this are added into the branch of Wine included with Proton or one not connected to it?

**Comment by Blayer98** [comment]
I wanted to also mention about some more issues I found while using the application, I decided not to initially include them in as I wanted the main focus to be on Linux generally and the saving issue for plugins, which seems to be happening every time you attempt to save one.

- The virtual keyboard cannot be opened while the Creation Kit windows are being used.

- The memory when loading Starfield.esm and setting it as an active file on Steam Deck will fill up to a point where the system freezes and a hard reboot is required (Hold Power button for 4 seconds)

- Regarding the last point, there have been reports that people using 16GB or even 8GB were apparently able to use the Creation Kit to an extent.

- When moving around or rotating the camera on the Render Window, the focus seems to be overridden by the other windows.

**Comment by Blayer98** [comment]
Just another addition regarding the use of CK on Steam Deck, it seems only 8-9GB of Starfield.esm can be loaded until the Steam Deck freezes and a hard reboot is required. Otherwise with only DefaultWorld usable and editable, 3-4GB is in use. Obviously this is a hardware issue, nothing to do with Proton, but I figured I’d mention it here as I’m sure people will try running the CK on less than minimum requirements like 16GB.

**Comment by ManApart** [comment]
I'm experiencing the same issue with crashes on save, so I appreciate the issue creation.

Fwiw, here's the [proton db entry](https://www.protondb.com/app/2722710) for this app, which includes the problem definition and related user specs from a number of people, including myself.

I loved using the TES creation kits back in the days when I was a windows user, and so I'm really hoping people are able to figure out the linux specific issues so I can use the SF CK on my linux box!

**Comment by Blayer98** [comment]
Just thought I’d pop back on and say that it *is* possible to load Starfield.esm on Steam Deck, but you need to increase the swapfile (stored at home/) to around 8GB or more (More would be much better). Here’s some instructions on that: https://www.reddit.com/r/SteamDeck/s/wjBwGHmjXp

You don’t need to disable the readonly mode in SteamOS either from my own testing, so that step can be ignored.

**Comment by Blayer98** [comment]
Another update: I checked the Skyrim Special Edition Creation Kit and found that saving appears to work just fine, overwriting also works fine, and the plugin's name show up in the title bar.  Loading and unloading the plugin worked there too, so there's a chance Starfield's CK does something different when saving plugins since the KERNEL32.CopyFile2 unimplemented function message did not show up in the log I generated for Skyrim's CK.

**Comment by ManApart** [comment]
In the starfield discord, someone said that they have it working after copying files into winecfg:
> I have been able to use Creation Kit fine on Linux in Proton. You need to add the dll files in your Creation Kit root to winecfg libraries as native, built-in for full function of Creation Kit. I noticed this originally when using Fallout and Skyrim's Creation Kit.

I'm attempting to replicate that by going to the CK folder, running winecfg, and adding the each of the DLL names from the Starfield folder. (I have no idea if that's the correct way to do this).
![image](https://github.com/user-attachments/assets/696b04c1-e7c5-498c-9ddb-1f07a1eed443)

**Comment by ManApart** [comment]
Running `protontricks -c winecfg 1716740` to attempt to do winecfg specific to the proton game gives me the error `Invalid file magic number. The appinfo.vdf version might not be supported by the current version of Protontricks - please check for updates.`. 

```
$ protontricks -V
protontricks (1.11.1)
```

**Comment by Blayer98** [comment]
> Running `protontricks -c winecfg 1716740` to attempt to do winecfg specific to the proton game gives me the error `Invalid file magic number. The appinfo.vdf version might not be supported by the current version of Protontricks - please check for updates.`.
> 
> ```
> $ protontricks -V
> protontricks (1.11.1)
> ```

Are you not running Creation kit through Steam? I would have done these steps on the Creation Kit’s ID, not Starfield’s ID. Also does this fix the issue regarding saving? The issue @kisak-valve saw in my proton log was that KERNEL32.dll.CopyFile2 was unimplemented.

**Comment by Blayer98** [comment]
Even with the winecfg additions, I'm still getting the same error and crash:

> wine: Call from 00006FFFFFFA2DC8 to unimplemented function KERNEL32.dll.CopyFile2, aborting

Someone would need to implement Kernel32's CopyFile2 function it seems and I'm not sure if any other games use this command.

Here's the Proton log from this session: 
[steam-2722710.log](https://github.com/user-attachments/files/16201613/steam-2722710.log)

On looking up CopyFile2, it seems it was a new addition in Windows 8 onwards according to this post:
https://stackoverflow.com/questions/66820467/error-copyfile2-could-not-be-located-in-the-dynamic-link-library-kernel32-dll-af
I guess this explains why it wasn't implemented yet because it was only added recently.

**Comment by ozoak** [comment]
By no means do I know what I'm doing with Wine or Proton, but from searching...

CopyFile2 was added to Wine in 9.3, Feb 2024:

https://gitlab.winehq.org/wine/wine/-/releases/wine-9.3

"Bugs fixed in 9.3 (total 23):
[#55897](https://bugs.winehq.org/show_bug.cgi?id=55897)  cpython 3.12.0 crashes due to unimplemented CopyFile2"

"Changes since 9.2:
Kartavya Vashishtha (1):
      kernelbase: Implement CopyFile2()."

The approved & merged fix referenced for the above linked bug:
https://gitlab.winehq.org/wine/wine/-/merge_requests/5020
https://source.winehq.org/git/wine.git/commit/1e348427994e5675ded3b1540a15a8af85cc8ebc

Now, right this minute, I may not know what to do with all that, however it looks like the work toward a resolution has already been done. Does that help someone with more current/advanced knowledge?

**Comment by ManApart** [comment]
> Are you not running Creation kit through Steam? I would have done these steps on the Creation Kit’s ID, not Starfield’s ID.

Yes, I'm running the CK through Steam. I had tried both SF and SF CK and just happened to copy the wrong id into the post above. Both apps give me the same error.

Ozoak, that's great to hear! Hopefully it's just a matter of time!

**Comment by Blayer98** [comment]
That’s great that it’s been implemented! I’m not sure when Valve decides to update Wine however. Currently Wine according to winecfg is 9.0.

According to the GitLab release dates, Wine 9.3 released on 23rd February while 9.0 released on 16th January. (Funny enough, Wine is already at 9.12 from 2 weeks ago lol)

From looking at Proton’s changelogs, Wine seems to get updated with every new major Proton version:
Proton 7.0-1 updated Wine to 7.0.
Proton 8.0-1 updated Wine to 8.0.
Proton 9.0-1 updated Wine to 9.0.

Based on this, it could be a while before this one fix gets added into Proton, although I could be completely wrong lol.

I just saw that Proton 9.0-1 released in May 2nd, so around 2 months ago from when Wine 9.3 released.

**Comment by Blayer98** [comment]
> > Are you not running Creation kit through Steam? I would have done these steps on the Creation Kit’s ID, not Starfield’s ID.
> 
> Yes, I'm running the CK through Steam. I had tried both SF and SF CK and just happened to copy the wrong id into the post above. Both apps give me the same error.
> 
> Ozoak, that's great to hear! Hopefully it's just a matter of time!

That’s odd, I use the version of Protontricks from Discover that launches it with -gui, but I haven’t seen that error before… I wouldn’t know why that’s happening sadly :(

**Comment by robotman40** [comment]
> Hello @Blayer98, `wine: Call from 00006FFFFFFA2DC8 to unimplemented function KERNEL32.dll.CopyFile2, aborting` looks like the line of interest from your Proton log.

I thought this was implemented awhile ago due to a bug with the shutil library in the Windows version of Python 3.12

**Comment by Blayer98** [comment]
> > Hello @Blayer98, `wine: Call from 00006FFFFFFA2DC8 to unimplemented function KERNEL32.dll.CopyFile2, aborting` looks like the line of interest from your Proton log.
> 
> I thought this was implemented awhile ago due to a bug with the shutil library in the Windows version of Python 3.12

The fix was added into Wine 9.3, but Wine in Proton is still at 9.0.

**Comment by ManApart** [comment]
FWIW I tried running ptorontricks from discover, specifically a new flathub install, and I get the same error.

That said, it's probably off topic of the current issue, but just posting here for max info. FWIW I was hoping to try to add all the DLs in the starfield folder to the protontricks for the CK just to vet the rumor that that makes it work...


```
Protontricks was closed due to the following error:

Traceback (most recent call last):
  File "/app/lib/python3.11/site-packages/protontricks/cli/util.py", line 159, in wrapper
    return cli_func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/protontricks/cli/main.py", line 277, in main
    proton_app = find_proton_app(
                 ^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/protontricks/steam.py", line 848, in find_proton_app
    tool_app = find_steam_compat_tool_app(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/protontricks/steam.py", line 609, in find_steam_compat_tool_app
    appinfo_sections = [
                       ^
  File "/app/lib/python3.11/site-packages/protontricks/steam.py", line 609, in <listcomp>
    appinfo_sections = [
                       ^
  File "/app/lib/python3.11/site-packages/protontricks/steam.py", line 531, in iter_appinfo_sections
    raise SyntaxError(
SyntaxError: Invalid file magic number. The appinfo.vdf version might not be supported by the current version of Protontricks - please check for updates.


=============

Please include this entire error message when making a bug report.
Log messages:

Running inside Flatpak sandbox, version 1.14.6.
Found Steam directory at /home/user/.steam/debian-installation
Using default Steam Runtime at /home/user/.steam/debian-installation/ubuntu12_32/steam-runtime
WINETRICKS environment variable is not available. Searching from $PATH.
Found 2 Steam library folders
Multiple compatdata directories found for app 1167630
Multiple compatdata directories found for app 1172470
Multiple compatdata directories found for app 584400
Multiple compatdata directories found for app 383980
Multiple compatdata directories found for app 1817070
Currently logged-in Steam user: 
Found 1 Steam shortcuts running using Steam compatibility tools
Using 'yad' as GUI provider
```

**Comment by ManApart** [comment]
With steam closed, I attempted to run the CK with WINEDLLOVERRIDES set for all the DLLs in the starfield folder:
```
WINEDLLOVERRIDES="bink2w64,DejaInsight.x64,flowchartx64,GFSDK_Aftermath_Lib.x64,libcrypto-1_1-x64,libeay32,libmysql,libssl-1_1-x64,libxess,NRD,NvLowLatencyVk,nvngx_dlss,nvngx_dlssg,nvngx_dlss_overridden,Qt5Core,Qt5Gui,Qt5Network,Qt5OpenGL,Qt5Sql,Qt5Svg,Qt5Widgets,QtAdvancedDocking,sfse_1_12_36,sl.common,sl.dlss,sl.dlss_g,sl.interposer,sl.nis,sl.nrd,sl.reflex,ssce5564,ssleay32,steam_api64,vcruntime140_1,WinPixEventRuntime,XCurl=n,b" steam steam://rungameid/2722710
```

I see a number of logs in the terminal output that indicate it's using those DLLs:
`File 'Z:\mnt\extradrive1\SteamLibrary\steamapps\common\Starfield\/sl.reflex.dll' is signed by NVIDIA and the signature was verified.`

Without loading all the game files, I try to save an esp.

I see the same unimplemented function:
`wine: Call from 0000000170062128 to unimplemented function KERNEL32.dll.CopyFile2, aborting`

I see a crash message that seems smushed with other logs:
```
Caching cursor image for , serial 13908, cache size = 9
CRASH (UNKNOWN ERROR:)Game 2722710 created interface STEAMAPPLIST_INTERFACE_VERSION001 / AppList
Game 2722710 created interface STEAMAPPS_INTERFACE_VERSION008 / Apps
```

**Comment by Blayer98** [comment]
The saving issue seems to have been fixed as of 31st July, the experimental-9.0 branch has implemented CopyFile2, and the changelog mentions this fix. I haven’t yet tested this out but the CK should now be able to save properly.

**Comment by Blayer98** [comment]
Can confirm, the saving issue has been fixed and works on Steam Deck. The only issue now with Experimental compared to older versions is that the editor stays on a black screen longer than Proton 8, but this should hopefully be in the logs already which were run on Proton Experimental. :)

**Comment by ManApart** [comment]
@Blayer98 , did you do anything specific to get it to work?

I made sure I was on proton experimental, verified files, and tried with no args, as well as the list of winedll overrides and in both cases still see the CK crash right after it seems to load up all the initial windows.

**Comment by Blayer98** [comment]
> @Blayer98 , did you do anything specific to get it to work?
> 
> I made sure I was on proton experimental, verified files, and tried with no args, as well as the list of winedll overrides and in both cases still see the CK crash right after it seems to load up all the initial windows.

I didn’t no, I left the CK installed on my Steam Deck, and I booted it back up a day ago and simply updated Proton Experimental which was already assigned onto the CK. :)

**Comment by ozoak** [comment]
> @Blayer98 , did you do anything specific to get it to work?
> 
> I made sure I was on proton experimental, verified files, and tried with no args, as well as the list of winedll overrides and in both cases still see the CK crash right after it seems to load up all the initial windows.

I've installed Mint 22, clean installed Steam, Starfield and the CK (and purged all prefix/compatdata), set to Proton Experimental, and still crash just after launch.

It might be that the differentiating component is the graphics drivers - I've got a Turing NV GPU (2070S). I analysed the proton debug log on v8 (where it crashes on save, but does load into the editor) vs the experimental debug, and I found a point of difference which was an issue which I resolved (the NV driver install had given me a second "unknown display" which was causing issues), but it still crashes shortly after loading all screens in. And I'm just not familiar with all the components (dx, vulkan, wine, proton, qt) and what is/isn't accepted 'warn' events in the log to know where to focus.

[steam-2722710.log](https://github.com/user-attachments/files/16504121/steam-2722710.log)
Given this issue was created specifically to not being able to save and appears it might be resolved, I should create a new issue I think.

(proton log attached for what it's worth)

**Comment by Blayer98** [comment]
It’s fine, just post any new issues inside this one otherwise the issues page gets cluttered with duplicates lol

Just double checking, is the log you’ve uploaded the one with the crashing upon application load?

**Comment by ozoak** [comment]
> It’s fine, just post any new issues inside this one otherwise the issues page gets cluttered with duplicates lol
> 
> Just double checking, is the log you’ve uploaded the one with the crashing upon application load?

Fair enough. And yep, that was the log from crash on load using experimental.

This appears to be the exception that brings it down:
```
11586.673:0124:0250:info:vkd3d-proton:dxgi_vk_swap_chain_init_sleep_state: Timer interval is 1.0 ms.
11586.673:0124:0250:warn:seh:dispatch_exception backtrace: --- Exception 0xc0000005.
11586.673:0124:0250:trace:seh:dispatch_exception code=c0000005 flags=0 addr=00006FFFF77ED61C ip=6ffff77ed61c
11586.673:0124:0250:trace:seh:dispatch_exception  info[0]=0000000000000001
11586.673:0124:0250:trace:seh:dispatch_exception  info[1]=0000000000000014
11586.673:0124:0250:warn:seh:dispatch_exception EXCEPTION_ACCESS_VIOLATION exception (code=c0000005) raised
```

**Comment by Blayer98** [comment]
I did notice this as well: proton:d3d12_pipeline_library_load_pipeline: Pipeline "Pipeline-34091305481" does not exist.

these Pipeline errors are constant in your log, and I don’t have this at all in mine…

**Comment by Blayer98** [comment]
Here's my logs for Proton Experimental & Proton 8:

[Proton8-StarfieldCK.log](https://github.com/user-attachments/files/16512049/Proton8-StarfieldCK.log)
[ProtonEXP-StarfieldCK.log](https://github.com/user-attachments/files/16512053/ProtonEXP-StarfieldCK.log)

@kisak-valve is there anything you can identify from these logs that are causing the hang upon quitting the program or the windows to stay black and not show anything for longer?

**Comment by ozoak** [comment]
> I did notice this as well: proton:d3d12_pipeline_library_load_pipeline: Pipeline "Pipeline-34091305481" does not exist.
> 
> these Pipeline errors are constant in your log, and I don’t have this at all in mine…

Yeah, I noticed those, so many. Though execution continues.
However d3d12core.dll is the first dll in the backtrace, so maybe that makes sense.

**Comment by ManApart** [comment]
FWIW Here's my proton experimental log as well:

[steam-2722710.log](https://github.com/user-attachments/files/16517283/steam-2722710.log)

**Comment by ManApart** [comment]
Just launched the new patch that came with vehicles.
Experimental: Same issue as before.
8.0-5: Same issue as before.

**Comment by ManApart** [comment]
Still seeing the same behavior after updating to Shattered Space.

**Comment by deveris256** [comment]
Fedora 42, Proton 10.0-2 (beta)

<img width="1968" height="1013" alt="Image" src="https://github.com/user-attachments/assets/f6f755c4-5926-4311-81c3-b12d38d49854" />

I was able to launch Starfield Creation Kit, launch a few windows such as galaxy view and the terrain block preview, and save the plugin too.

Launched in the following order:
- Mod Organizer 2, which was installed using some installer script, was launched through Steam Proton
- Through Mod Organizer 2, I launched Creation Kit.

Haven't tested it extensively, but that's the first glance on it.

**Comment by ManApart** [comment]
FWIW the CK seems to be working fine for me now. I've created a very small esp/esm and seen them both work in game.

**Comment by ManApart** [comment]
In Constructible Object, I am unable to Select Form and then pick a Created Object. The form accepts my selected item but the Constructible Object is not updated.

NOTE: I can work around by viewing the constructible object in the inspector and changing it there.

**Comment by yak3d** [comment]
I can get it to launch, and everything mostly works but selecting an item in the render window will cause it to freeze. It seems to be spending a ton of time in a thread that calls into `winevulkan.dll`.

**Comment by yak3d** [comment]
So I ended up fixing the clicking on render window bug, but it required a combination of a cachyOS fork of proton + these launch opts: `PROTON_VKD3D_HEAP=1 VKD3D_CONFIG=descriptor_heap %command%`. It looks like there was a bug that came with the 595 NVIDIA driver, that even 595.71 doesn't fix.

When clicking on items in the render window without this fix, the NVIDIA driver reports:
```
NVRM: Xid (PCI:0000:01:00): 109, pid=44101, name=CreationKit.exe, channel 0x00000023, errorString CTX SWITCH TIMEOUT, Info 0x1fc02f
```

**Comment by Blayer98** [comment]
[steam-2722710.zip](https://github.com/user-attachments/files/29225749/steam-2722710.zip)

When editing a map marker, if you attempt to add a new reference in the LinkedRef tab on the map marker's properties window, and then select the reference via the render window (Cell window), and select ok, the application will crash, I think by a access violation error?

UPDATE: Crash also occurs when manually entering the details normally done when selecting reference via render window.

<img width="396" height="426" alt="Image" src="https://github.com/user-attachments/assets/8cb816ec-8f74-481c-a51c-6739770f1a77" />

**Comment by Blayer98** [comment]
Here's the dmp file the Creation Kit generated after the crash. I can confirm that the issue also occurs on any object when attempting to add a LinkedRef to any object, including making any edits to a LinkedRef. Deleting a LinkedRef does not cause a crash however.

[CreationKit_06-23-10-37.zip](https://github.com/user-attachments/files/29246883/CreationKit_06-23-10-37.zip)

This bug is still occurring on Proton Experimental after switching from 10.4. Using ProtonGE 10-34 also did not work and a crash occurred on that Proton version too.

**Comment by ManApart** [comment]
I ran into this crash as well and ended up installing windows on an older machine as it was the only way I could find to work around it.

**Comment by Blayer98** [comment]
There are also bugs with the Creation Kit when it comes to Wwise integration when using Proton:

Make sure to load up Starfield.esm first! These bugs occurred on Steam Deck via Proton Experimental and may/may not occur on other Linux distros.

My Plugin Name: LinuxPluginBugs.esp

Attempting to record any dialogue via the Edit Response window fails (For instance I added new dialogue to the "DialogueConstellation" quest scene and in Sarah Morgan's Phase 1.), no .wav or .wem is created at Data\Sound\Voice\LinuxPluginBugs.esp\NPCFSarahMorgan\ and the file path "LinuxPluginBugs.esp\NPCFSarahMorgan\" was not created either. I had to manually create the file path myself or let the RoboVoice option do it for me.

The specific error I got was this: "AUDIO:    Error acquiring FileSink interface (HResult: -2147467262)" The Configure Mic option does recognise that my Steam Deck has a microphone, but is unable to use it.

~~If the file path does exist and a .wem file with the right file name exists, it cannot be previewed in the Edit Response window as the CK thinks it doesn't exist when it does. Starfield however, will recognise it and play it as normal when playtesting the mod. (Make sure the voiced dialogue isn't being said by the player, as they don't play at all in game!) Any .wem files stored in Starfield.esm or any other master file should be able to be previewed fine as in my own testing.~~

EDIT: If a .wem file with the correct file name exists in the plugin folder for audio, e.g; "Data\Sound\Voice\LinuxPluginBugs.esp\NPCFSarahMorgan\00ADDABF.wem", the Edit Response window may say that the wem file exists but it cannot be previewed as it thinks the length of the audio is 0 seconds. If you go to File > Data, and then reload both Starfield.esm and the plugin you selected, and go back to the dialogue and the edit response window, you should be able to preview it. At least, this fixed the preview issue for me when adding new voice lines into my plugin/master file, and the seconds displayed was 0.


Attempting to Process Local Voice WAVs (Audio tab) when a wav file with the right file name exists (in my case this was "NPCFSarahMorgan\003F2A04.wav"), will not work. Voice Starvation warnings will appear as well. This should convert any .wav files into .wem, but this didn't occur.

Attempting to select RoboVoice in the Edit Response window will display this error on Steam Deck: (This should generate the NPC folders and the .esp folder for you regardless if RoboVoice gives an error or not!)

<img width="394" height="401" alt="Image" src="https://github.com/user-attachments/assets/3eeec5ec-d8bb-4af8-b1d7-9676819ea6bd" />

TL;DR, the following features of the CK work/don't work on Linux with minor/major issues so far: (Anything not listed works as expected to my knowledge!)

✅❔❌
Loading Data ✅ (Make sure you have enough RAM. If on Steam Deck, update zram-generator.conf to use 14.5GB. Don't load big cities!)
Wwise Integration ❌ (Process Local Voice WAVs fails with Voice Starvation warnings, auto creating .wem files fails.)

Cell Editing:
Object Properties ✅&❌ (Adding/Editing LinkedRefs crash the CK, otherwise everything appears to work fine.)
Loading Cells ✅ (In Cells that contain multiple Cells at once, moving around in the Cell window, and a new Cell loads in whilst moving, focus is taken away and given to Cell View, otherwise works fine.)

Editing Dialogue:
Recording dialogue ❌ (Fails with: "AUDIO:    Error acquiring FileSink interface (HResult: -2147467262)")
RoboVoicer ❌ (Fails with: System.PlatformNotSupportedException: No voice installed on the system or none available with the current security setting.)
Previewing voice files ❔ (Not at first. If Seconds in Edit Response window says 0, but wem is listed as Y, reload both Starfield.esm and the current plugin, this should fix the issue.)
Adding animations to dialogue ❌ (Crashes the CK.)

**Comment by Blayer98** [comment]
New crash found: Attempting to add or change any animations in a scene's dialogue will cause a crash on Linux.

[steam-2722710.zip](https://github.com/user-attachments/files/29516942/steam-2722710.zip)
This zip file also contains the dmp file generated by the Creation Kit.

**Comment by madewokherd** [comment]
Those exceptions and traces are all from Xalia. There are no mscoree traces from other processes.

**Comment by madewokherd** [comment]
Oops, didn't realize which issue tracker I was on.

**Comment by Blayer98** [comment]
Okay, so here is what I've been able to find thanks to Visual Studio Code, but first, this is what I did to analyse the dmp file:

- Install Visual Studio Code through Discover
- Run VSC, and go to the Extensions tab
- Search for Minidump Parser, and install it
- Select Minidump Parser in the activity tab (left hand row)
- Choose "Open New Dump File" ands select the .dmp file.

Here's the proton log to compare this dump to:
[steam-2722710.zip](https://github.com/user-attachments/files/29564622/steam-2722710.zip)

This is the analysed minidump after adding a Linked Ref to an object and clicking Ok.
```
=== CRASH SUMMARY ===
Crash Reason  : EXCEPTION_ACCESS_VIOLATION_READ
Crash Address : 0xffffffff96e06d88
Crashing Thread: 5
Crashing Module: CreationKit.exe
OS            : Windows NT
CPU           : amd64
Uptime        : not available
Stack Quality : LOW (1331/1544 frames from stack scanning) - majority of frames found by stack scanning, results may be inaccurate
==================================================
Operating system: Windows NT
                  10.0.19045 
CPU: amd64
     family 23 model 145 stepping 0
     8 CPUs

GPU: UNKNOWN

Crash reason:  EXCEPTION_ACCESS_VIOLATION_READ
Crash address: 0xffffffff96e06d88
Process uptime: not available

Thread 5 (crashed)
 0  CreationKit.exe + 0x1daf92c
    rax = 0xffffffff96e06d80   rdx = 0x0000000000000000
    rcx = 0x0000000000000000   rbx = 0x0000000000000000
    rsi = 0x0000000000109db0   rdi = 0x0000000000109dc0
    rbp = 0x0000000000109e01   rsp = 0x0000000000109c90
     r8 = 0x0000000000000150    r9 = 0x0000000000000000
    r10 = 0x0000000000000008   r11 = 0x0000000000000202
    r12 = 0xffffffffffffffff   r13 = 0x00000001744fc1c0
    r14 = 0x0000000000109d98   r15 = 0x0000000000109ea0
    rip = 0x0000000141daf92c
    Found by: given as instruction pointer in context

Thread 0
 0  ntdll.dll + 0xea94
    rax = 0x00006fffffedd8b0   rdx = 0x0000000000000000
    rcx = 0x0000000000103790   rbx = 0x0000000000000000
    rsi = 0x0000000000000001   rdi = 0x0000000758edbae0
    rbp = 0x0000000000108890   rsp = 0x0000000000108848
     r8 = 0x0000000000104b70    r9 = 0x00006fffffedd8b0
    r10 = 0x0000000000000000   r11 = 0x0000000000000246
    r12 = 0x0000000000108d10   r13 = 0x0000000000000000
    r14 = 0x000000007ffc0000   r15 = 0x0000000000000000
    rip = 0x00006ffffff3ea94
    Found by: given as instruction pointer in context
 1  kernelbase.dll + 0x7238f
    rax = 0x00006fffffedd8b0   rdx = 0x0000000000000000
    rcx = 0x0000000000103790   rbx = 0x0000000000000000
    rsi = 0x0000000000000001   rdi = 0x0000000758edbae0
    rbp = 0x0000000000108890   rsp = 0x0000000000108850
     r8 = 0x0000000000104b70    r9 = 0x00006fffffedd8b0
    r10 = 0x0000000000000000   r11 = 0x0000000000000246
    r12 = 0x0000000000108d10   r13 = 0x0000000000000000
    r14 = 0x000000007ffc0000   r15 = 0x0000000000000000
    rip = 0x00006fffffc6238f
    Found by: simulating a return from leaf function
 2  ntdll.dll + 0xeb54  [confidence: LOW - stack scanning ⚠]
    rbp = 0x0000000000108890   rsp = 0x0000000000108858
    rip = 0x00006ffffff3eb54
    Found by: stack scanning
 3  kernelbase.dll + 0x581c9  [confidence: LOW - stack scanning ⚠]
    rbp = 0x0000000000108890   rsp = 0x0000000000108860
    rip = 0x00006fffffc481c9
    Found by: stack scanning
 4  CreationKit.exe + 0x4e268cf  [confidence: LOW - stack scanning ⚠]
    rbp = 0x0000000000108890   rsp = 0x0000000000108870
    rip = 0x0000000144e268cf
    Found by: stack scanning
 5  CreationKit.exe + 0x17c5b288  [confidence: LOW - stack scanning ⚠]
    rbp = 0x0000000000108890   rsp = 0x0000000000108880
    rip = 0x0000000157c5b288
    Found by: stack scanning
``` 
Based on the vkd3d-proton errors when the Object Properties window is open, I can only presume this issue is being caused by vkd3d-proton? The problem I'm stuck with however, is that Creation KIt itself crashes, so I'm kinda stumped... I also have no way of live debugging the CK and trying to find out what processes are in use when it does crash...

**Comment by Blayer98** [comment]
> New crash found: Attempting to add or change any animations in a scene's dialogue will cause a crash on Linux.
> 
> [steam-2722710.zip](https://github.com/user-attachments/files/29516942/steam-2722710.zip) This zip file also contains the dmp file generated by the Creation Kit.


POTENTIAL WORKAROUND FOUND!!
When attempting to add a dialogue animation, select the Animation button under “Dialogue Animation”, and browse for the animation you want. Depending on the NPC you’re adding an animation to, make sure the Actor matches the NPC or the animation you add in may not play.

**DO NOT** double click on the animation’s file name as the CK will crash. Instead, type it out into a text editor like Kate, copy it, and then exit the animation selection window. Now, select “Send Anim Event” and paste the animation’s file name. Press Ok, and if that’s the only change you’re making, select Minor Edit as not doing so may change the Voice file’s hex value.

I haven’t found a workaround for adding Linked Refs yet in the CK, but I think you could can add one/multiple in xEdit.
